pppd: Remove the "record" option

On many embedded systems there is not enough space to record PPP session
information to the permanent storage, therfore remove this option.

Signed-off-by: Jo-Philipp Wich <jo@mein.io>

--- a/pppd/pppd.h
+++ b/pppd/pppd.h
@@ -326,7 +326,6 @@ extern int	holdoff;	/* Dead time before
 extern bool	holdoff_specified; /* true if user gave a holdoff value */
 extern bool	notty;		/* Stdin/out is not a tty */
 extern char	*pty_socket;	/* Socket to connect to pty */
-extern char	*record_file;	/* File to record chars sent/received */
 extern bool	sync_serial;	/* Device is synchronous serial device */
 extern int	maxfail;	/* Max # of unsuccessful connection attempts */
 extern char	linkname[MAXPATHLEN]; /* logical name for link */
--- a/pppd/tty.c
+++ b/pppd/tty.c
@@ -145,7 +145,7 @@ char	*disconnect_script = NULL; /* Scrip
 char	*welcomer = NULL;	/* Script to run after phys link estab. */
 char	*ptycommand = NULL;	/* Command to run on other side of pty */
 bool	notty = 0;		/* Stdin/out is not a tty */
-char	*record_file = NULL;	/* File to record chars sent/received */
+static  char	*const record_file = NULL;	/* File to record chars sent/received */
 int	max_data_rate;		/* max bytes/sec through charshunt */
 bool	sync_serial = 0;	/* Device is synchronous serial device */
 char	*pty_socket = NULL;	/* Socket to connect to pty */
@@ -201,8 +201,10 @@ option_t tty_options[] = {
       "Send and receive over socket, arg is host:port",
       OPT_PRIO | OPT_DEVNAM },
 
+#if 0
     { "record", o_string, &record_file,
       "Record characters sent/received to file", OPT_PRIO },
+#endif
 
     { "crtscts", o_int, &crtscts,
       "Set hardware (RTS/CTS) flow control",
